backend/api/schedule/types.py

```diff
@@ -35,6 +37,25 @@ class ScheduleItem:
     language: Language
     audience_level: Optional["AudienceLevel"]
 
+    @strawberry.field
+    def has_free_spot(self, info) -> bool:
+        return self.capacity_left > 0
+
+    @strawberry.field
+    def is_booked(self, info) -> bool:
+        user = info.context["request"].user
+
+        if not user.is_authenticated:
+            return False
+
+        return ScheduleItemBooking.objects.filter(
+            user=user, schedule_item=self
+        ).exists()
+
+    @strawberry.field
+    def can_book(self, info) -> bool:
+        return self.allows_booking and self.capacity_left > 0
```

Comment:
we could reuse the resolver 😊

```suggestion
    def can_book(self, info) -> bool:
        return self.allows_booking and self.has_free_spot(info)
```

Reply:
we could also move the resolver to be a property on the model, but that's minor

Reply:
moved it to the model